### PR TITLE
[REVIEW] Support for DURATION type

### LINKED ISSUE #1467: add support for duration type
this is very similar to the issue in https://github.com/BlazingDB/blazingsql/issues/1451

We want to add support for being able to use duration types in operations

### PR DESCRIPTION
**Pull request by Christian8491**
This PR closes #1467.  This one is related to BlazingDB/blazingsql-testing-files#29  PR which adds `INTERVAL` tests.

This PR enables the use of many `INTERVAL` sql operations like: 

sql1 = `"select duration_col  + INTERVAL '1'  SECOND, duration_col  - INTERVAL '1'  HOUR  from interval_table"`
sql2 = `"select INTERVAL '125 10' DAY(3) TO HOUR from interval_table"`
sql3 = `"select duration_col + INTERVAL '2 01:03:10' DAY TO SECOND from interval_table"`

Due to the large range of `DURATION/INTERVAL`  sql queries, I just have tested some of them, you can check this list of enabled queries  from the `intervalTest.py` file. 

This PR does not enables `DURATION` type for `data_parsers` as they are still experimental and some new e2e test should be provided also. 

Sql operatios over `MONTH` and `YEAR` (with `DURATION` type) are currently not supported (`SECOND`, `MINUTE`, `HOUR`, `DAY` are) .

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by Christian8491** [comment]
rerun tests

**Comment by Christian8491** [comment]
rerun tests

**Comment by Christian8491** [comment]
rerun tests

**Comment by Christian8491** [comment]
rerun tests

**Comment by Christian8491** [comment]
@williamBlazing  could you reopen this PR pointing to the "new`" branch-21.06 `

File: tests/BlazingSQLTest/EndToEndTests/TestSuites/intervalSuite.yaml [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+  TEST_57:
+    SQL: |
+        (select i_id, i_duration_s from interval_table limit 36)
+            union all 
+        (select i_id, i_duration_ms from interval_table limit 4)
+
+
+  # NOTE: all this query works for Blazing but not for Drill
+  # TEST_58
+  #   select i_duration_s, INTERVAL '1' SECOND as col_to_sum,
```
Comment (williamBlazing) [inline tests/BlazingSQLTest/EndToEndTests/TestSuites/intervalSuite.yaml:265]: do these work for Spark?

File: tests/BlazingSQLTest/EndToEndTests/TestSuites/intervalSuite.yaml [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  # Join with DURATION same unit
+  TEST_52:
+    SQL: |
+        select r1.i_id + 1 as r1_id, r1.i_duration_s as r1_duration_s,
+                r2.i_duration_ms as r2_duration_ms
+            from interval_table r1 inner join interval_table r2
+            on r1.i_duration_s = r2.i_duration_s
+            order by r1.i_id limit 70       
+
+  # Join with DURATION same unit
```
Comment (williamBlazing) [inline tests/BlazingSQLTest/EndToEndTests/TestSuites/intervalSuite.yaml:231]: join with DURATION diff unit

**Review by williamBlazing — APPROVED** [review]
Have some small questions, but otherwise looks good

File: tests/BlazingSQLTest/EndToEndTests/TestSuites/intervalSuite.yaml (hunk 2 above)
Comment (Christian8491) [reply tests/BlazingSQLTest/EndToEndTests/TestSuites/intervalSuite.yaml]: updated this comment.

File: tests/BlazingSQLTest/EndToEndTests/TestSuites/intervalSuite.yaml (hunk 1 above)
Comment (Christian8491) [reply tests/BlazingSQLTest/EndToEndTests/TestSuites/intervalSuite.yaml:265]: All these tests were just faced again Drill due to the schema. Spark does not support `timedelta64[X]` variations as Drill does.

**Comment by Christian8491** [comment]
rerun tests

**Comment by Christian8491** [comment]
rerun tests

**Pull request merged**